instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/LoggerBuilderE2ETests.kt

```diff
@@ -0,0 +1,170 @@
+        }
+        logger.sendRandomLog(testMethodName, forge)
+    }
+
+    /**
+     * apiMethodSignature: Logger#Builder#fun setNetworkInfoEnabled(Boolean): Builder
+     */
+    @Test
+    fun logs_logger_builder_network_info_enabled() {
+        val testMethodName = "logs_logger_builder_network_info_enabled"
+        measureLoggerInitialize {
+            logger = Logger.Builder().setNetworkInfoEnabled(true).build()
+        }
+        logger.sendRandomLog(testMethodName, forge)
+    }
+
+    /**
+     * apiMethodSignature: Logger#Builder#fun setNetworkInfoEnabled(Boolean): Builder
+     */
+    @Test
+    fun logs_logger_builder_network_info_disabled() {
+        val testMethodName = "logs_logger_builder_network_info_disabled"
+        measureLoggerInitialize {
+            logger = Logger.Builder().setNetworkInfoEnabled(false).build()
+        }
+        logger.sendRandomLog(testMethodName, forge)
+    }
+
+    /**
+     * apiMethodSignature: Logger#Builder#fun setServiceName(String): Builder
+     */
+    @Test
+    fun logs_logger_builder_set_service_name() {
+        val testMethodName = "logs_logger_builder_set_service_name"
+        measureLoggerInitialize {
+            logger = Logger.Builder().setServiceName(CUSTOM_SERVICE_NAME).build()
+        }
+        logger.sendRandomLog(testMethodName, forge)
+    }
+
+    /**
+     * apiMethodSignature: Logger#Builder#fun setLoggerName(String): Builder
+     */
+    @Test
+    fun logs_logger_builder_set_logger_name() {
+        val testMethodName = "logs_logger_builder_set_logger_name"
+        measureLoggerInitialize {
+            logger = Logger.Builder().setLoggerName(CUSTOM_LOGGER_NAME).build()
+        }
+        logger.sendRandomLog(testMethodName, forge)
+    }
+
+    /**
+     * apiMethodSignature: Logger#Builder#fun setBundleWithRumEnabled(Boolean): Builder
+     */
+    @Test
+    fun logs_logger_builder_bundle_with_rum_enabled() {
+        val testMethodName = "logs_logger_builder_bundle_with_rum_enabled"
+        measureLoggerInitialize {
+            logger = Logger.Builder().setBundleWithRumEnabled(true).build()
+        }
+        val viewKey = forge.anAlphabeticalString()
+        GlobalRum.get().startView(viewKey, forge.anAlphabeticalString())
+        logger.sendRandomLog(testMethodName, forge)
+        GlobalRum.get().stopView(viewKey)
+    }
+
+    /**
+     * apiMethodSignature: Logger#Builder#fun setBundleWithRumEnabled(Boolean): Builder
+     */
+    @Test
+    fun logs_logger_builder_bundle_with_rum_disabled() {
+        val testMethodName = "logs_logger_builder_bundle_with_rum_disabled"
+        measureLoggerInitialize {
+            logger = Logger.Builder().setBundleWithRumEnabled(false).build()
+        }
+        val viewKey = forge.anAlphabeticalString()
+        GlobalRum.get().startView(viewKey, forge.anAlphabeticalString())
+        logger.sendRandomLog(testMethodName, forge)
+        GlobalRum.get().stopView(viewKey)
+    }
+
+    companion object {
```

Comment:
You're missing test for `setBundleWithTraceEnabled`

Reply:
Nope, as I explained last week during the daily I could not find a way for now to assert this test case. The `dd.trace_id` or `dd.span_id` are not queryable and not even visible in the Log event attributes dashboard.

Reply:
Even though you can't assert that the `dd.trace_id` and `dd.span_id`  are present, you can still assert that the log exist, right? A partial assertion is always better than no assertions at all